common-lib/src/main/scala/com/gu/mediaservice/lib/imaging/im4jwrapper/ImageMagick.scala

```diff
@@ -18,9 +21,20 @@ object ImageMagick {
   def addImage(source: File) = (new IMOperation()) <| { op => { op.addImage(source.getAbsolutePath) }}
   def quality(op: IMOperation)(qual: Double) = op <| (_.quality(qual))
   def stripMeta(op: IMOperation) = op <| (_.strip())
+  def stripProfile(op: IMOperation)(profile: String) = op <| (_.p_profile(profile))
   def addDestImage(op: IMOperation)(dest: File) = op <| (_.addImage(dest.getAbsolutePath))
   def crop(op: IMOperation)(b: Bounds): IMOperation = op <| (_.crop(b.width, b.height, b.x, b.y))
   def profile(op: IMOperation)(profileFileLocation: String): IMOperation = op <| (_.profile(profileFileLocation))
+  def thumbnail(op: IMOperation)(width: Int): IMOperation = op <| (_.thumbnail(width))
   def scale(op: IMOperation)(dimensions: Dimensions): IMOperation = op <| (_.scale(dimensions.width, dimensions.height))
+  def format(op: IMOperation)(definition: String): IMOperation = op <| (_.format(definition))
+
   def runConvertCmd(op: IMOperation): Future[Unit] = Future((new ConvertCmd).run(op))
+  def runIdentifyCmd(op: IMOperation): Future[List[String]] = Future {
+    val cmd = new IdentifyCmd()
```

Comment:
Are you presuming that the identify command infers the colorspace from (a) some heuristic on the image data itself, or (b) uses image metadata?

When looking at this previously I was defeated by the ImageMagick source code being impenetrable to me. I _think_ testing indicated that it is in fact (a). If this is the case then images with incorrect or missing metadata will still fail this test.

To confirm which situation is occurring you could intentionally strip all metadata from an image and then run the identify command to see whether it can correctly infer the colorspace.